server/src/connection.ts

```
@@ -0,0 +1,60 @@
+import { auth, FirebaseError } from "firebase-admin";
+import { Socket } from "socket.io";
+
+// Represents a Socket.io event.
+enum Event {
+    CONNECTION = 'connection',
+    DISCONNECT = 'disconnect',
+    UPDATE = 'update',
+}
+
+// Represents a user socket.
+interface UserSocket extends Socket {
+    // A user's uuid.
+    uuid?: string;
+}
+
+// Represents an authentication error.
+class AuthenticationError extends Error {
+    constructor(message: string) {
+        super(message);
+        this.name = 'AuthenticationError';
+    }
+}
+
+// Connects all clients with the same uuid to a room and listens for updates from other clients.
+function connect(socket: UserSocket): void {
+    const uuid = socket.uuid;
+    console.log(`${uuid} has connected`);
+    if (uuid) {
```

I think this checks if `uuid` is null. In what cases will `uuid` not exist?

---

if the client does not send UUID correctly

---

Isn't that what the error handling on `authenticate()` is for? I do not see any benefits of having this.

---

Actually, this is way worse than it seems, it's completely swallowing the error message and it doesn't even inform the client what went wrong. At the very least, the server should disconnect the user.

---

oh, my mistake, I misinterpreted the question. You are right, there are no cases that uuid does not exist

---

If that's the case, is there a point to have that if check?